homeassistant/components/devolo_home_control/switch.py

```diff
@@ -0,0 +1,155 @@
+
+
+class DevoloSwitch(SwitchDevice):
+    """Representation of an Awesome Light."""
+
+    def __init__(self, homecontrol, device_instance, element_uid):
+        """Initialize an devolo Switch."""
+        self._device_instance = device_instance
+
+        # Create the unique ID
+        self._unique_id = element_uid
+
+        self._homecontrol = homecontrol
+        self._name = self._device_instance.itemName
+        self._available = self._device_instance.is_online()
+
+        # Get the brand and model information
+        self._brand = self._device_instance.brand
+        self._model = self._device_instance.name
+
+        self._binary_switch_property = self._device_instance.binary_switch_property.get(
+            self._unique_id
+        )
+        self._is_on = self._binary_switch_property.state
+
+        if hasattr(self._device_instance, "consumption_property"):
+            self._consumption = self._device_instance.consumption_property.get(
+                self._unique_id.replace("BinarySwitch", "Meter")
+            ).current
+        else:
+            self._consumption = None
+
+        self.subscriber = None
+
+    async def async_added_to_hass(self):
+        """Call when entity is added to hass."""
+        self.subscriber = Subscriber(
+            self._device_instance.itemName, callback=self.update
+        )
+        self._homecontrol.publisher.register(
+            self._device_instance.uid, self.subscriber, self.update
+        )
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of the switch."""
+        return self._unique_id
+
+    @property
+    def device_info(self):
+        """Return the device info."""
+        return {
+            "identifiers": {(DOMAIN, self._device_instance.uid)},
+            "name": self.name,
+            "manufacturer": self._brand,
+            "model": self._model,
+        }
+
+    @property
+    def device_id(self):
+        """Return the ID of this switch."""
+        return self._unique_id
+
+    @property
+    def name(self):
+        """Return the display name of this switch."""
+        return self._name
+
+    @property
+    def should_poll(self):
+        """Return the polling state."""
+        return False
+
+    @property
+    def is_on(self):
+        """Return the state."""
+        return self._is_on
+
+    @property
+    def current_power_w(self):
+        """Return the current consumption."""
+        return self._consumption
+
+    @property
+    def available(self):
+        """Return the online state."""
+        return self._available
+
+    def turn_on(self, **kwargs):
+        """Switch on the device."""
+        self._is_on = True
+        self._binary_switch_property.set_binary_switch(state=True)
+
+    def turn_off(self, **kwargs):
+        """Switch off the device."""
+        self._is_on = False
+        self._binary_switch_property.set_binary_switch(state=False)
+
+    def update(self, message=None):
```

Comment:
`update` is a reserved method name on the entity. It's used to fetch new state data by the default polling mechanism in home assistant and to fetch initial state data during entity addition. Please use another name, since we don't want to use those things.